### PR TITLE
DEV-1372 | Contributor portal links expired on arrival

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This PR fixes a bug with contributor portal links, albeit for reasons other than what we initially thought were causing problem.

Specifically, it marks safe the magic link that gets sent to the django email template and ensures that the email address in the magic link is HTML quoted. 

Before, when it wasn't marked safe, the `&email=` in the magic link query params was getting changed to `&amp;email=`, which in turn meant that in the API, there was no `email` value arriving in the contributor verify request (which on the SPA side takes the values from the query params and posts them in request body). Additionally, for emails like "foo+bar.com", the "+" was a problem.

Note that while this PR fixes the issue with contributor portal access, it appears to reveal a new bug whereby contributions don't show up in the contributor portal even when they are recorded in NRH admin and on stripe. For instance, I made [this donation](https://dev-1372.revengine-review.org/nrhadmin/contributions/contribution/26/change/), but when I log in as that contributor, I don't see any donations.

Behavior can be verified here: https://billypenn-dev-1372.revengine-review.org/contributor

#### What's this PR do?

- Minimal amount of cleanup. Removes a path through view that would never occur, and removes unused `NoSuchContributorError`
- Uses `mark_safe` on value for magic link so query params don't get escaped.
- Uses `quote_plus` on the value for email in the magic link
- Adds a "beefy" test spanning two request cycles and parametrized with "standard" and "edge case" email addresses to prove correctness of the solution.

#### Why are we doing this? How does it help us?

Allows contributor portal to be accessed.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Make a contribution, then go to https://billypenn-dev-1372.revengine-review.org/contributor and request magic link for same email you used for contribution. Link should arrive and should allow you to access contributor portal (even though you probably won't see any contributions).

Be sure to test emails with `+` signs in them. (_Update from Daniel_)

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

[DEV-1372](https://news-revenue-hub.atlassian.net/jira/software/c/projects/DEV/boards/3?modal=detail&selectedIssue=DEV-1372)

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

[DEV-2171](https://news-revenue-hub.atlassian.net/browse/DEV-2171)

### CODE REVIEW

**Review by njharman — APPROVED** [review]
lgtm; 99% coverage

Did not fill out pre-review checklist

Guess we have no way to automate pulling link from email and seeing that it 200's against "test" server.

**Review by x110dc — CHANGES_REQUESTED** [review]
Following the testing instructions I contributed (https://billypenn-dev-1372.revengine-review.org/nrhadmin/contributions/contribution/37/change/) but when I followed the email link to the portal it said it was expired:

Screen Shot 2022-08-02 at 3 13 17 PM

File: apps/api/tests/test_views.py [hunk 1]
```diff
@@ -88,6 +95,55 @@ def test_delete_removes_auth_cookie(self):
...
+    assert response.status_code == 200
+    assert response.json()["contributor"]["email"] == email
+    jwt_data = jwt.decode(response.cookies["Authorization"].value, settings.SECRET_KEY, algorithms="HS256")
+    assert jwt_data["token_type"] == "access"
+    assert jwt_data["ctx"] == LONG_TOKEN
+    assert jwt_data["exp"] > jwt_data["iat"]
+    assert jwt_data["exp"] > int(time())
+    contributor = Contributor.objects.get(email=email)
+    assert jwt_data["contrib_id"] == str(contributor.uuid)
+
```
Comment (x110dc) [inline apps/api/tests/test_views.py:145]: That is one sophisticated, beautiful test!

**Review by x110dc — APPROVED** [review]
